build/Jenkinsfile

```diff
@@ -1,52 +1,80 @@
 #!/usr/bin/env groovy
 
-def gitCommit
-node {
-    stage('Checkout') {
-        checkout scm
-        gitCommit = sh(returnStdout: true, script: 'git rev-parse HEAD').trim()
+pipeline {
+    agent {
+        docker {
+            image 'mattermost/mattermost-build-webapp:v5.2.0'
+        }
     }
-    stage('Check Style') {
-        sh 'make check-style'
-    }
-    stage('Unit/Component Tests') {
-        sh 'make test'
-    }
-    stage('Build') {
-        sh 'make build'
-        sh 'make package'
+
+    environment {
+        GIT_COMMIT = sh(returnStdout: true, script: 'git rev-parse HEAD').trim()
     }
-    stage('Push to S3') {
-        step([$class: 'S3BucketPublisher', dontWaitForConcurrentBuildCompletion: false, entries: [[
-            bucket: 'releases.mattermost.com/mattermost-webapp/${BRANCH_NAME}',
-            excludedFile: '',
-            flatten: true,
-            gzipFiles: false,
-            keepForever: false,
-            managedArtifacts: false,
-            noUploadOnFailure: true,
-            selectedRegion: 'us-east-1',
-            showDirectlyInBrowser: false,
-            sourceFile: '*.tar.gz',
-            storageClass: 'STANDARD',
-            uploadFromSlave: false,
-            useServerSideEncryption: false,
-            userMetadata: [[key: 'Cache-Control', value: 'no-cache']]
-        ], [
-            bucket: "releases.mattermost.com/mattermost-webapp/commit/${gitCommit}",
-            excludedFile: '',
-            flatten: true,
-            gzipFiles: false,
-            keepForever: false,
-            managedArtifacts: false,
-            noUploadOnFailure: true,
-            selectedRegion: 'us-east-1',
-            showDirectlyInBrowser: false,
-            sourceFile: '*.tar.gz',
-            storageClass: 'STANDARD',
-            uploadFromSlave: false,
-            useServerSideEncryption: false,
-            userMetadata: [[key: 'Cache-Control', value: 'no-cache']]
-        ]], profileName: 'Releases', userMetadAta: []])
+
+    stages {
+        stage('Install') {
+            steps {
+                sh 'make .npminstall'
+            }
+        }
+
+        stage('Check Style') {
+            steps {
+                sh 'make check-style'
+            }
+        }
+
+        stage('Test') {
+            steps {
+                sh 'make test'
+            }
+            post {
```

Comment:
This deviates slightly from what I did in mattermost-redux, in that if the build failed prior to this stage, we didn't actually want to do the `junit` step. I'll patch up mattermost-redux later.